addon/src/lib/events.js

```diff
@@ -1,87 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-/* global Components, Services */
-
-import PubSub from "pubsub-js";
-import allTopics from "./topics";
-
-import { log } from "./utils";
-import { registerWebExtensionAPI } from "./webExtension";
-
-const { utils: Cu } = Components;
-Cu.import("resource://gre/modules/Services.jsm");
-const { addObserver, removeObserver, notifyObservers } = Services.obs;
-
-const eventsTopics = (...args) => allTopics("bootstrap", "events", ...args);
-
-const EVENT_PREFIX = "testpilot::";
-
-const observers = {};
-
-export async function startupEvents() {
-  log("startupEvents");
-  registerWebExtensionAPI("observeEventTopic", observeEventTopic);
-  registerWebExtensionAPI("notifyEventTopic", notifyEventTopic);
-  addObserver(observeIdleDaily, "idle-daily", false);
-}
-
-export async function shutdownEvents() {
-  log("shutdownEvents");
-  removeObserver(observeIdleDaily, "idle-daily");
-  Object.keys(observers).forEach(unobserveEventTopic);
-}
-
-export function notifyEventTopic({ subject, topic, payload }) {
-  notifyObservers(subject, `${EVENT_PREFIX}${topic}`, JSON.stringify(payload));
-}
-
-export function observeEventTopic(eventTopic) {
-  const observer = new EventObserver(eventTopic);
-  observers[eventTopic] = observer;
-  observer.register();
-}
-
-export function unobserveEventTopic(eventTopic) {
-  observers[eventTopic].unregister();
-  delete observers[eventTopic];
-}
-
-function observeIdleDaily() {
-  PubSub.publishSync(eventsTopics("idle-daily"));
```

Comment:
Worth noting here is that [code listening for this idle-daily event](https://github.com/mozilla/testpilot/search?utf8=%E2%9C%93&q=idle-daily&type=) is what [sends the daily telemetry ping](https://github.com/mozilla/testpilot/blob/6f67e2ad6a7d2b7e600bbefe60b98f1b4bf16494/addon/src/webextension/lib/metrics.js#L100). So, this is also somewhat related to #3177 in that it will stop the daily ping altogether